pkg/clustermanager/eksa_mover.go

```diff
@@ -0,0 +1,165 @@
+type MoverOpt func(*Mover)
+
+// Mover applies the cluster spec to the management cluster and waits
+// until the changes are fully reconciled.
+type Mover struct {
+	log                                        logr.Logger
+	clientFactory                              ClientFactory
+	applyClusterTimeout, waitForFailureMessage time.Duration
+	retryBackOff                               time.Duration
+}
+
+// NewMover builds an Mover.
+func NewMover(log logr.Logger, clientFactory ClientFactory, opts ...MoverOpt) *Mover {
+	m := &Mover{
+		log:                   log,
+		clientFactory:         clientFactory,
+		applyClusterTimeout:   applyClusterSpecTimeout,
+		waitForFailureMessage: waitForFailureMessageErrorTimeout,
+		retryBackOff:          retryBackOff,
+	}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m
+}
+
+// WithMoverNoTimeouts disables the timeout for all the waits and retries in management upgrader.
+func WithMoverNoTimeouts() MoverOpt {
+	return func(a *Mover) {
+		maxTime := time.Duration(math.MaxInt64)
+		a.applyClusterTimeout = maxTime
+		a.waitForFailureMessage = maxTime
+	}
+}
+
+// WithMoverApplyClusterTimeout allows to configure how long the mover retries
+// to apply the objects in case of failure.
+// Generally only used in tests.
+func WithMoverApplyClusterTimeout(timeout time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.applyClusterTimeout = timeout
+	}
+}
+
+// WithMoverRetryBackOff allows to configure how long the mover waits between requests
+// to update the cluster spec objects and check the status of the Cluster.
+// Generally only used in tests.
+func WithMoverRetryBackOff(backOff time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.retryBackOff = backOff
+	}
+}
+
+// Move applies the cluster's namespace and spec without checking for reconcile conditions.
+func (m *Mover) Move(ctx context.Context, spec *cluster.Spec, fromCluster, toCluster *types.Cluster) error {
+	m.log.V(3).Info("Moving the cluster object")
+	err := retrier.New(
+		m.applyClusterTimeout,
+		retrier.WithRetryPolicy(retrier.BackOffPolicy(m.retryBackOff)),
+	).Retry(func() error {
+		var err error
+		fromClient, err := m.clientFactory.BuildClientFromKubeconfig(fromCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to read cluster objects")
+		}
+
+		toClient, err := m.clientFactory.BuildClientFromKubeconfig(toCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to move cluster objects")
+		}
+
+		// read the cluster from bootstrap
+		cluster := &v1alpha1.Cluster{}
+		if err := fromClient.Get(ctx, spec.Cluster.Name, spec.Cluster.Namespace, cluster); err != nil {
+			return errors.Wrapf(err, "getting cluster from %s", fromCluster.Name)
+		}
+
+		// pause cluster on bootstrap
+		cluster.PauseReconcile()
+		if err := fromClient.ApplyServerSide(ctx, defaultFieldManager, cluster, kubernetes.ApplyServerSideOptions{ForceOwnership: true}); err != nil {
```

Comment:
I would use `Update` here (assuming it's implemented by this client)
IMO this usecase is a classic read+update with optimistic locking (using resource version)